### PR TITLE
labels: add stringlabels version of StableHash

### PR DESCRIPTION
**Pull request by bboreham**
This leaves the slice-based implementation alone, and adds an implementation for `-tags stringlabels` which
inlines the `Range()` implementation to avoid adding overhead.

### CODE REVIEW

File: model/labels/sharding_string.go
Comment (colega) [inline model/labels/sharding_string.go:1]: Just like in: https://github.com/prometheus/prometheus/pull/12328
I'd suggest (ask? :pray:) calling this with a different suffix than `_string.go`, like `_stringlabels.go`.

**Review by colega — APPROVED** [review]
Thank you, @bboreham, this LGTM, please consider renaming the file.

I hope you don't mind that I pushed here a test and a CI definition to test with `stringlabels`.

File: model/labels/sharding_string.go
Comment (bboreham) [reply model/labels/sharding_string.go:1]: No objection; it parallels `labels_string.go` but that could be changed upstream.

Comment (colega) [reply model/labels/sharding_string.go:1]: Yes, I also sent a request to change that upstream: https://github.com/prometheus/prometheus/pull/12328

Comment (colega) [reply model/labels/sharding_string.go:1]: Renamed here: https://github.com/grafana/mimir-prometheus/pull/503/commits/1f14d27d4006bf55d3e554247251fc4dbd513c20

**Pull request merged**